y/iterator.go

```diff
@@ -20,38 +20,42 @@ package y
 // Meta field.
 type ValueStruct struct {
 	Meta     byte
-	UserMeta byte
+	UserMeta []byte
 	Value    []byte
 
 	Version uint64 // This field is not serialized. Only for internal usage.
 }
 
 // EncodedSize is the size of the ValueStruct when encoded
 func (v *ValueStruct) EncodedSize() uint16 {
-	return uint16(len(v.Value) + 2) // meta, usermeta.
+	return uint16(len(v.Value) + len(v.UserMeta) + 2) // meta
 }
 
 // Decode uses the length of the slice to infer the length of the Value field.
 func (v *ValueStruct) Decode(b []byte) {
 	v.Meta = b[0]
-	v.UserMeta = b[1]
-	v.Value = b[2:]
+	if b[1] != 0 {
```

Comment:
We don't need `if`, and make `2+b[1]` a local variable is easier to read.

Reply:
Without `if`, the result of `v.UserMeta` will not be `nil`. I want it be `nil` if `len(v.UserMeta) == 0`, because if user doesn't set `UserMeta` it should be `nil`.

Reply:
Then we need to set it to `nil` explicitly, the `ValueStruct` may be reused.